File: Try_GameEngine5/src/com/example/try_gameengine/framework/IChessBoard.java
package com.example.try_gameengine.framework;

import java.util.List;


import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Point;
import android.graphics.PointF;

public interface IChessBoard extends IDrawEvent{
	public void createLines();
	public void createPoints();
	public Point newPoint(Float x, Float y);
	public int[][] getAllExistPoints();
	public void setAllExistPoints(int[][] allExistPoints);
	public int getLineDistance();
	public void setPlayersBySquential(List<IChessPlayer> playersBySquential);
	public PointF getScreenXYByChessPoint(Point p);
}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/IChessBoard.kt
package com.example.try_gameengine.framework

import android.graphics.Point
import android.graphics.PointF

interface IChessBoard : IDrawEvent {
    fun createLines()
    fun createPoints()
    fun newPoint(x: Float, y: Float): Point
    fun getAllExistPoints(): Array<IntArray?>
    fun setAllExistPoints(allExistPoints: Array<IntArray?>)
    fun getLineDistance(): Int
    fun setPlayersBySquential(playersBySquential: MutableList<IChessPlayer?>?)
    fun getScreenXYByChessPoint(p: Point): PointF
}
